File: financeiro-api/src/main/java/br/com/iftm/financeiroapi/model/service/EntryService.java
package br.com.iftm.financeiroapi.model.service;

import br.com.iftm.financeiroapi.model.domain.Entry;
import br.com.iftm.financeiroapi.model.exceptions.BusinessException;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.util.List;

@Service
public interface EntryService {

    Entry save(Entry entry) throws IOException, BusinessException;

    void delete(String id) throws IOException;

    List<Entry> findAll() throws IOException, BusinessException;

    Entry findById(String id) throws IOException, BusinessException;

}

File: financeiro-api-kotlin/src/main/kotlin/br/com/iftm/financeiroapi/model/service/EntryService.kt
package br.com.iftm.financeiroapi.model.service

import br.com.iftm.financeiroapi.model.domain.Entry
import org.springframework.stereotype.Service
import java.math.BigDecimal

@Service
interface EntryService {

    fun save(entry: Entry): Entry

    fun delete(id: String)

    fun findAll(): List<Entry>

    fun findById(id: String): Entry?

    fun findByCategoryName(categoryName: String): List<Entry>

    fun generateEntries()

    fun sumEntriesByCategory(categoryName: String): BigDecimal
}